src/Hazelcast.Net.Tests/Clustering/FailoverTests.cs

```diff
@@ -0,0 +1,492 @@
+﻿// Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Threading.Tasks;
+using Hazelcast.Clustering;
+using Hazelcast.Core;
+using Hazelcast.DistributedObjects;
+using Hazelcast.Partitioning;
+using Hazelcast.Security;
+using Hazelcast.Testing;
+using Hazelcast.Testing.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Moq;
+using NUnit.Framework;
+using Hazelcast.Exceptions;
+using Hazelcast.Configuration;
+
+namespace Hazelcast.Tests.Clustering
+{
+    [Category("enterprise")]
+    [Timeout(120_000)]
+    internal class FailoverTests : MultipleClusterRemoteTestBase
+    {
+        private IDisposable HConsoleForTest()
+
+
+            => HConsole.Capture(options => options
+                .ClearAll()
+                .Configure().SetMinLevel()
+                .Configure<HConsoleLoggerProvider>().SetMaxLevel()
+                .Configure<Failover>().SetPrefix("FAILOVER").SetMaxLevel()
+                .Configure<FailoverTests>().SetPrefix("TEST").SetMaxLevel()
+            );
+
+        const string clusterAName = "clusterA";
+        const string clusterBName = "clusterB";
+
+        const string clusterAKey = "keyForClusterA";
+        const string clusterAData = "dataForClusterA";
+
+        const string clusterBKey = "keyForClusterB";
+        const string clusterBData = "dataForClusterB";
+
+        private static ClusterState MockClusterState(HazelcastOptions options)
+        {
+            var mockpartitioner = Mock.Of<Partitioner>();
+            return new ClusterState(options, "clusterName", "clientName", mockpartitioner, new NullLoggerFactory());
+        }
+
+        [Test]
+        public async Task TestFailsWhenEnabledWithNoCluster()
+        {
+            var opt = new HazelcastFailoverOptionsBuilder()
+                .Build();
```

Comment:
I feel like it would be better to validate the config here while building it. WDYT?

Reply:
the `HazelcastFailoverOptionsBuilder` should not build invalid options, so yes the validation should be part of the builder and the exception should be thrown when building - separation of concerns.

Reply:
This will be addressed after the release